lib/block_view_stake.go

```diff
@@ -1338,8 +1341,10 @@ func (bav *UtxoView) _disconnectStake(
 		bav._setStakeEntryMappings(operationData.PrevStakeEntries[0])
 	}
 
-	// Restore the PrevGlobalStakeAmountNanos.
-	bav._setGlobalStakeAmountNanos(operationData.PrevGlobalStakeAmountNanos)
+	// Restore the PrevGlobalActiveStakeAmountNanos if the PrevValidatorEntry was active.
+	if prevValidatorEntry.Status() == ValidatorStatusActive {
```

Comment:
we could just check if `PrevGlobalActiveStakeAmountNanos` is non-nil. I generally prefer that as opposed to relying on a different field. The prevs generally means "revert to this state when disconnecting".